### PR TITLE
fix up pandas stubs to make tests pass

### PR DESCRIPTION
**Pull request by Dr-Irv**
I decided to spend some time to see how many of the tests that are now in the repo can be fixed by fixing up various stubs.

This PR fixes most of them except for the following:

- ~~`divmod` - it wasn't clear to me how to get a direct call to `divmod` to work for pandas given what is in the typeshed~~
- ~~`Interval.length.total_seconds()` - not sure how to handle this.  If the interval corresponds to timestamps, then this is valid~~
- `Series.values` - not sure how to handle this.  `Series.values` can return an `ExtensionArray` or `nd.ndarray`

So when I run `pyright` on this version, I'm down to 4 errors, all related to `Series.values` .

This probably includes some reformatting due to black.

### CODE REVIEW

**Comment by Dr-Irv** [comment]
@gramster - I think you're the one to look at this, and hopefully it helps given all the work you have been doing.

**Comment by Dr-Irv** [comment]
With respect to `Series.values`, I have created an issue in the `pandas-stubs` repo to get their thoughts:  https://github.com/VirtusLab/pandas-stubs/issues/142

**Comment by gramster** [comment]
This is great, thanks Irv! I have a PR which removes stubs where we add no value over the current inline annotations, so I need to see if there are files you touched that I removed, and possibly bring them back. But I should have some time this afternoon to get through this.

File: pandas/core/frame.pyi [hunk 1]
```diff
@@ -722,8 +874,8 @@ class DataFrame(NDFrame):
     @overload
     def count(self, axis: AxisType = ..., level: None = ..., numeric_only: _bool = ...) -> Series[Dtype]: ...
     def nunique(self, axis: AxisType = ..., dropna=True) -> Series[Dtype]: ...
-    def idxmax(self, axis: AxisType, skipna: _bool = ...) -> Series[Dtype]: ...
-    def idxmin(self, axis: AxisType, skipna: _bool = ...) -> Series[Dtype]: ...
+    def idxmax(self, axis: AxisType = 0, skipna: _bool = ...) -> Series[Dtype]: ...
```
Comment (gramster) [inline pandas/core/frame.pyi:877]: This should be '= ...' on line above and below

**Comment by gramster** [comment]
There's a lot of noise introduced by Black here but I think this is okay to go apart from the '= 0' I pointed out. Will need to merge this before my PR that moves the stubs and makes them partial. I'll wait for you to fix the one comment and then we can merge this. My PR moves to using Black too so that will help in future.

**Review by gramster — APPROVED** [review]

**Comment by Dr-Irv** [comment]
> There's a lot of noise introduced by Black here but I think this is okay to go apart from the '= 0' I pointed out. Will need to merge this before my PR that moves the stubs and makes them partial. In the interests of time I will merge this and make that fix in my PR.

OK.  Wasn't sure if you wanted me to fix it, so I created #150

**Pull request merged**